### ISSUE TITLE
Popover: can't make it fit inside a mobile screen

### ISSUE THREAD

**Issue by KKoukiou** [patternfly/patternfly-react#5993]
I have trouble making the popover component adjust it's width to fit inside a mobile screen. That said passing `hasAutoWidth` or not, the popovers always make it out of the screen like seen in the sceenshot bellow.
![Screen Shot 2021-07-07 at 18 13 21](https://user-images.githubusercontent.com/14921356/124794319-2b7db080-df4f-11eb-9e0b-e97471eba272.png)

**Comment by KKoukiou** [comment]
The screenshot is with IPhone SE IOS, from the firefox available mobile views.

**Comment by tlabaj** [comment]
cc @mcoker is this due to the min-width?

**Comment by mcoker** [comment]
@KKoukiou the `hasAutoWidth` is working for me - https://www.patternfly.org/v4/components/popover/react/width-auto/

With that variation, you could set `--pf-c-popover--MaxWidth: min(300px, 90%);`, which will allow the popover to be 300px when there is room, and 90% when there isn't.

>  is this due to the min-width?

@tlabaj as far as I can tell, it looks like it may be a combination of both that, and that the popover is appearing at the center of the element that triggers it. Can popper know to offset the position so that the popover is within the bounds of the viewport?

**Comment by KKoukiou** [comment]
@mcoker thanks for checking this.:) 
I don't know what I am doing wrong, but https://www.patternfly.org/v4/components/popover/react/width-auto/ popover escapes out of the screen in mobile mode for me, also when setting `--pf-c-popover--MaxWidth: min(300px, 90%);`

**Comment by mcoker** [comment]
@KKoukiou no problem! I'm using the mobile emulator in chrome dev tools. After setting `--pf-c-popover--MaxWidth`, I do have to trigger the popover to open again before I see the change. Are you testing in dev tools or some other way?

<img width="1682" alt="Screen Shot 2021-07-12 at 9 58 35 AM" src="https://user-images.githubusercontent.com/35148959/125315374-adc5f600-e2fc-11eb-8c3f-5c48c4ac4fbd.png">

**Comment by KKoukiou** [comment]
Right, was doing the same in firefox, if it's works for you just ignore and let someone else tests it. It will be something on my side, will check it at some later point :)

**Comment by KKoukiou** [comment]
@mcarrano unfortunately this is not fixed. I am really not sure how this is working for you. Here is a reproducer with the suggested custom CSS workaround.

https://codesandbox.io/s/popover-auto-width-v7x12

In mobile mode the popover is still half hidden.

**Comment by garrett** [comment]
I tried to figure out a solution with CSS, but it's being set by JavaScript. One way to figure out placement for the left transition is to calculate the width of the popover element (+ padding) versus the page.

So you could do something like this pseudocode:

```js
isConstrainedByMobile = pageWidth - popoverWidth < (popoverPaddingX * 2)
```

So if the page is wide enough, it would be like this:

1200 - 200 < 16*2 = 1000 < 32 = false

240 - (240 * .9) < 32 = 24 < 32 = true

(This is assuming 90% width hardcoded above a an example. The JS would actually get the pixel width for the popover and the page width via JS for calculating the above.)

And then do something based on `isConstrainedByMobile`, like change the translation offset X to 0. (We don't want the popover shifting offscreen to the left _or_ right.)

```js
translate(isConstrainedByMobile ? 0 : offsetX, offsetY)
```

And then you'd want to then apply half of the offsetX to the arrow, either with adjusting left or a transform so the arrow is properly pointing to the origin.

**Comment by mcoker** [comment]
@KKoukiou sorry for the delay. It looks like in my previous example, the popover happened to fit my mobile viewport, but the popover's positioning will move it off screen.

I forked your codesandbox with the `min()` function referenced above - it also needs the popover's `min-width: none` - https://codesandbox.io/s/popover-auto-width-forked-5jnr28?file=/custom.css

Another thing that may help, though it doesn't solve the problem, is using diagonal positioning, which I added to the codesandbox above.

**Comment by mcarrano** [comment]
@KKoukiou can we close this? Let us know if you need anything else.

**Comment by garrett** [comment]
> I forked your codesandbox with the min() function referenced above - it also needs the popover's min-width: none - https://codesandbox.io/s/popover-auto-width-forked-5jnr28?file=/custom.css

Thanks!

This works for me:

![Screenshot from 2022-03-08 09-29-17](https://user-images.githubusercontent.com/10246/157197632-132b566f-a5c5-4cc7-99e3-c3005b6be9da.png)


> @KKoukiou can we close this? Let us know if you need anything else.

Shouldn't this be the default behavior, instead of manually adjusting the CSS like this?  We should probably keep this issue open and resolve it by default in a PR.

**Comment by KKoukiou** [comment]
I agree with @garrett that this should be the default behevior when auto positioning is chosen.
